tripal/tripal_jbrowse
# Upgrade version 4.x to Drupal 10

Issue #76 

Tripal JBrowse 4.x has been in dire need of an upgrade to Drupal 10 since Drupal 9 has been officially deprecated. The following changes have been implemented:

1. The parameter `core_version_requirement` has been updated to `^10` in the info.yml
2. The Dockerfile was updated to use the latest version of Tripal on Drupal version: `10.1.x-dev`
3. The README was updated to reference the `drupal` directory instead of `drupal9`
4. `composer.json` was updated to require at least php 8.1, drupal 10.0 and tripal 4.x-dev (NOTE: This has NOT been tested yet, but I'll revisit this when a new version of tripal is released)

Additional updates to the Dockerfile were made for easier development, including:
- installing and prepping chado 
- importing general and genomic content types 
- enabling the tripal jbrowse module

Because of changes in the past couple of months that were made to the Tripal Docker, the autocompletes for `organism` and `genome assembly` on the Create JBrowse Instance page did not work. Those have now been updated to move away from using the outdated bio_data terms.

Lastly, I removed the JBrowse Management submodule from this repository for the time being, as the code was from the Tripal 3 version and made it hard to debug the above issue. I will revisit creating the module in the future when I reach that point in development.

### Testing
1. Clone this repository and switch to the branch `g0.76-drupal10update`
2. Build the docker using the following commands:
``` 
docker build . --tag=tripal_jbrowse:latest
docker run -dit --name=tripaljbrowse --publish=80:80 --volume=`pwd`:/var/www/drupal/web/modules/contrib/tripal_jbrowse tripal_jbrowse:latest
docker exec tripaljbrowse service postgresql restart
```
3. Go to http://localhost and log in using the credentials in the [Tripal Docker docs](https://tripaldoc.readthedocs.io/en/latest/install/docker.html#development-site-information)
5. Make sure that the Tripal JBrowse modules are already enabled at `Adminstration > Manage > Extend`
6. Go to `Administration > Tripal > Content > +Add Tripal Content` to manually create an Organism and Analysis
7. Create a JBrowse instance by going to the `JBrowse Instances` tab on the Tripal Content page.
8. Fill out the form using the Organism and Analysis just created and click Submit
